packages/@aws-cdk/assert/lib/assertions/have-output.ts

```diff
@@ -0,0 +1,82 @@
+import {Assertion} from "../assertion";
+import {StackInspector} from "../inspector";
+
+class HaveOutputAssertion extends Assertion<StackInspector> {
+  constructor(private readonly outputName?: string, private readonly exportName?: string, private outputValue?: any) {
+    super();
+  }
+
+  public get description(): string {
+    return `output\
+${this.outputName ? ' with name ' + this.outputName : ''}\
+${this.exportName ? ' with export name ' + this.exportName : ''}\
+${this.outputValue ? ' with value ' + JSON.stringify(this.outputValue) : ''}\
+`;
+  }
+
+  public assertUsing(inspector: StackInspector): boolean {
+    if (!this.outputName && !this.exportName) {
```

Comment:
This is a *very late* time to verify this. This should be done right in the constructor.